pontoon/settings/base.py

```diff
@@ -50,7 +54,29 @@ def path(*args):
 SESSION_COOKIE_HTTPONLY = os.environ.get('SESSION_COOKIE_HTTPONLY', 'True') != 'False'
 SESSION_COOKIE_SECURE = os.environ.get('SESSION_COOKIE_SECURE', 'True') != 'False'
 
-SITE_URL = os.environ.get('SITE_URL', 'http://localhost:8000')
+APP_URL_KEY = 'APP_URL'
+
+# For the sake of integration with Heroku, we dynamically load domain name
+# From the file that's set right after the build phase.
+if os.environ.get('HEROKU_DEMO') and not os.environ.get('SITE_URL'):
+    def _site_url():
+        from django.contrib.sites.models import Site
+        from django.core.cache import cache
+
+        app_url = cache.get(APP_URL_KEY)
+
+        # Sometimes data from cache is flushed, We can't do anything about that.
+        if not app_url:
+            app_url = "https://{}".format(Site.objects.get(pk=1).domain)
+            cache.set(APP_URL_KEY, app_url)
+
+        return app_url
+
+    SITE_URL = lazy(_site_url, str)()
+else:
+    SITE_URL = os.environ.get('SITE_URL', 'http://localhost:8000')
+
+BROWSERID_AUDIENCES = [SITE_URL]
```

Comment:
We no longer need `BROWSERID_AUDIENCES`. It was used by Persona and was removed from master.